services/autoscaling/src/simcore_service_autoscaling/modules/dask.py

```diff
@@ -162,9 +193,13 @@ async def get_worker_used_resources(
     def _get_worker_used_resources(
         dask_scheduler: distributed.Scheduler,
     ) -> dict[str, dict]:
-        used_resources = {}
+        used_resources: dict[str, dict] = {}
```

Comment:
Q: I wonder, is there a way to pass `worker_url` to the function running in the scheduler (AFAIU `_get_worker_userd_resources`) instead of listing them all and filtering in here? i.e. can you do the filtering remotely?

MINOR: `_get_worker_used_resources` suggested to me a single worker but here you are listing all resources per worker instead

Reply:
yes sorry, this comes from different iterations. and I agree with you this could be further optimized. will look into it